### ISSUE TITLE
Tests break when using Generator Return Expressions from PHP 7.0

### ISSUE THREAD

**Issue by mallardduck** [rskuipers/php-assumptions#15]
I just wanted to create this issue here for visibility for anyone who might have the same issue. Due to an issue in the [nikic/PHP-Parser](
https://github.com/nikic/PHP-Parser) package if you use self-executing anonymous functions the tests break and throw an error.

Full details can be seen here: nikic/PHP-Parser#316

**Comment by rskuipers** [comment]
@mallardduck Thank you for the heads-up! I'll be following the issue to see if we can resolve this :)

**Comment by mallardduck** [comment]
It looks like the other package owner commented and provided some details. From a quick look at the code of both projects it looks like some changes may be necessary to get PHP7 parsing working.

I don't have time now, but perhaps over the weekend I'll be able to take a look into it. If I can figure it out and get something working I'll throw a PR your way; unless you beat me to it.

**Comment by mallardduck** [comment]
@rskuipers so I submitted a pull request that fixes my issue. there's a bit of a catch though.

I've never used prophesy in my unit testing and I actually started using this package only last week. So needless to say I wasn't able to fix tests completely. I ran into a few issues related to this.

The first being that after updating nikic/php-parser I was getting errors related to prophecy-phpunit, specifically ProphecyTestCase. From that I learned that this was a deprecated package and I removed it, replacing with prophecy.

Then after realizing I got different results when using vendors phpunit version I started fixing those issues. In that process I updated phpunit to 5.x which drops support for php 5.5, should be fine as it's EOL though.

(TL;DR:)So all in all, the package is updated for php 7.x support; however I was unable to fix all unit tests and in the process of updating I removed support for PHP 5.5.
